pytest_tank_test/tk-testengine/engine.py

```diff
@@ -9,9 +9,38 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
-from sgtk.platform import Engine
+import sgtk
 
 
-class TestEngine(Engine):
+class TestEngine(sgtk.platform.Engine):
+    """
+    Test engine.
+
+    The engine will initialize a QApplication if possible right before
+    applications start registering themselves so it looks as if they
+    are running within a GUI environment.
+    """
+
+    def pre_app_init(self):
+        try:
+            app = sgtk.platform.qt.QtGui.QApplication.instance()
+            self._app = app or sgtk.platform.qt.QtGui.QApplication([])
+        except Exception:
+            # This will fail if Qt is not available.
+            self._app = None
+
+        if self._app:
+            self._initialize_dark_look_and_feel()
+
+    @property
+    def app(self):
+        """
+        The QtGui.QApplication instance, if available.
```

Comment:
Tests that launch the test engine will generally store it in `self.engine`, which then makes the QApplication very easy to reach: `self.engine.app` instead of `sgtk.platform.qt.QtGui.QApplication.instance()`